packages/e2e-playwright/tests/card/expiryDate/card.expiryDatePolicies.hidden.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { test, expect } from '../../../pages/cards/card.fixture';
+import { SYNCHRONY_PLCC_NO_DATE, TEST_CVC_VALUE, ENCRYPTED_CARD_NUMBER, ENCRYPTED_EXPIRY_DATE, ENCRYPTED_SECURITY_CODE } from '../../utils/constants';
+import LANG from '../../../../lib/src/language/locales/en-US.json';
+
+const PAN_ERROR = LANG['error.va.sf-cc-num.02'];
+const DATE_INVALID_ERROR = LANG['error.va.sf-cc-dat.01'];
+const DATE_EMPTY_ERROR = LANG['error.va.sf-cc-dat.04'];
+const CVC_ERROR = LANG['error.va.sf-cc-cvc.01'];
+
+test.describe('Test how Card Component handles hidden expiryDate policy', () => {
+    test('#1 how UI & state respond', async ({ cardExpiryDatePoliciesPage }) => {
+        const { card, page } = cardExpiryDatePoliciesPage;
+
+        await card.isComponentVisible();
+
+        // Fill number to provoke binLookup response
+        await card.typeCardNumber(SYNCHRONY_PLCC_NO_DATE);
+
+        // UI reflects that binLookup says expiryDate is hidden
+        await expect(card.expiryDateField).not.toBeVisible();
+
+        await card.typeCvc(TEST_CVC_VALUE);
+
+        // Card seen as valid
+        let cardValid = await page.evaluate('window.card.isValid');
+        await expect(cardValid).toEqual(true);
+
+        // Clear number
+        await card.deleteCardNumber();
+
+        // UI is reset
+        await expect(card.expiryDateField).toBeVisible();
+
+        // Card seen as invalid
+        cardValid = await page.evaluate('window.card.isValid');
+        await expect(cardValid).toEqual(false);
```

Comment:
Is there another way to check if card is invalid/valid instead of inspecting the window object?

Reply:
Not that I can think of. 
The test needs access to the actual _component_ that is initialised i.e. in `/app/src/pages/Cards/Cards.js` so that it can access the component's state
I don't know how else you could pass this to a test?